### ISSUE TITLE
Proper handling or errors raised in Unity#componentDidMount 

### ISSUE THREAD

**Issue by DorianGrey** [elraccoone/react-unity-webgl#115]
**Is your feature request related to a problem? Please describe.**
The `componentDidMount` lifecycle of the `Unity` component is asynchronous:
https://github.com/elraccoone/react-unity-webgl/blob/5a0c3c14093434a320cea8e25b6f10a54750fa18/source/components/unity.ts#L42

As such, errors raised in it resp. causing the promise to fail cannot be handled via a usual `ErrorBoundary` component and thus regularly end up in an `unhandledrejection` event that can only be handled globally.

**Describe the solution you'd like**
Regularly, errors that are raised by async lifecycles should be handled via a state and cause proper reactions, either in the component itself or its parent (or both). Thus, the component should catch such errors properly instead of letting them bubble up to the global level.
There are at least two options to actually respond to such an error in extension to catching it:
1. Notify the parent using the `error` message on the event listener on the `UnityContext` to be provided. Might not be intended to use this one that way, though.
2. Let the parent define some kind of `renderFallback` function that receives the error and returns the `ReactNode` to be rendered instead, like:
```typescript
export default interface IUnityProps {
  // ... other properties as already defined atm.
  renderFallback?: (err: any) => ReactNode;
}
```

**Describe alternatives you've considered**
Currently, I'm using inheritance on TS level to implement the error handling and trigger event invocation so that the parent component can react to the error properly (see option (1) as described above), like:

```typescript
import OldUnity from "react-unity-webgl";

interface UnityState {
  mountError?: Error;
}

export class Unity extends OldUnity {
  state: UnityState = {};
  async componentDidMount(): Promise<void> {
    try {
      await super.componentDidMount();
    } catch (err) {
      // Publish the error so that the surrounding component can
      // respond to it.
      const errorMsg: InitError = {
        type: "init" as const,
        error: err,
      };
      this.props.unityContext.invokeEventListener("error", errorMsg);
      this.setState({
        mountError: err,
      });
    }
  }

  render(): ReactNode {
    if (this.state.mountError) {
      return null;
    }
    return super.render();
  }
}
```

**Additional context**
The easiest way to trigger this errornous behavior is to provide invalid or unreachable URLs to any of the properties required by `UnityContext`.

**Comment by jeffreylanters** [comment]
Hi DorianGrey! Thank you for your details report. That's a good find, I will implement this in the next patch release.

**Comment by DorianGrey** [comment]
I would have provided a PR already if I could have made an actual decision about which options sounds better resp. is more reasonable, especially if the parent component requires to perform some additional task when detecting an error raised by the `Unity` component, like removing loading progress indicators.
* While the `renderFallback` version is quite concise and simple, it would take additional workarounds or tricky steps to trigger additional behavior in the parent component.
* While the event listener is simple to wire with additional responding behavior in the parent, this version appears to be some kind of misuse of the event system; or at least errors emitted this way should be specific values so that the parent can separate them from other errors potentially forwarded this way during runtime.

**Comment by jeffreylanters** [comment]
Issue should be fixed in `8.0.2`
